client/verta/verta/_deployment/endpoint.py

```diff
@@ -76,9 +77,67 @@ def _get_json_by_path(cls, conn, workspace, path):
         # TODO: iterate through response.json().get('endpoints', []) to find matching creator_request.path
 
     def update(self, run, strategy):
-        raise NotImplementedError
-        # TODO: check if isinstance(run, experimentrun.ExperimentRun)
-        # TODO: check if isinstance(strategy, deployment._UpdateStrategy)
+        if not isinstance(run, experimentrun.ExperimentRun):
+            raise TypeError("run must be an ExperimentRun")
+
+        if not issubclass(type(strategy), _UpdateStrategy):
+            raise TypeError("strategy must be an object from verta.deployment.strategies")
+
+        stage_id = self._get_or_create_stage()
+
+        # Create new build:
+        url = "{}://{}/api/v1/deployment/workspace/{}/builds".format(
+            self._conn.scheme,
+            self._conn.socket,
+            self.workspace
+        )
+        response = _utils.make_request("POST", url, self._conn, json={"run_id": run.id})
+        _utils.raise_for_http_error(response)
+        build_id = response.json()["id"]
+
+        # Update stages with new build
+        url = "{}://{}/api/v1/deployment/workspace/{}/endpoints/{}/stages/{}/update".format(
+            self._conn.scheme,
+            self._conn.socket,
+            self.workspace,
+            self.id,
+            stage_id
+        )
+        response = _utils.make_request("PUT", url, self._conn, json=strategy._as_build_update_req_body(build_id))
+        _utils.raise_for_http_error(response)
+
+    def _get_or_create_stage(self, name="production"):
+        if name == "production":
+            # Check if a stage exists:
+            url = "{}://{}/api/v1/deployment/workspace/{}/endpoints/{}/stages".format(
+                self._conn.scheme,
+                self._conn.socket,
+                self.workspace,
+                self.id
+            )
+            response = _utils.make_request("GET", url, self._conn, params={})
+
+            _utils.raise_for_http_error(response)
+            response_json = response.json()
+
+            if response_json["stages"]:
+                return response_json["stages"][0]["id"]
+
+            # no stage found:
+            return self._create_stage("production")
+        else:
+            raise NotImplementedError # currently not supported other stages
```

Comment:
I've just realized it would be helpful to include this in the error message
```suggestion
            raise NotImplementedError("currently not supported other stages")
```

Reply:
Didn't even know this is a thing 😲 Will fix.

Reply:
Fixed.